src/lib/pubkey/x25519/donna.cpp

```diff
@@ -345,13 +345,14 @@ void fmonty(uint64_t result_two_q_x[5],
 * information.
 */
 inline void swap_conditional(uint64_t a[5], uint64_t b[5], uint64_t c[5], uint64_t d[5], uint64_t iswap) {
-   const uint64_t swap = 0 - iswap;
+   const auto swap = CT::Mask<uint64_t>::expand(iswap);
```

Comment:
Nit-picky, but should't `swap_conditional` take a `CT::Choice` for `iswap` here?

Reply:
A `Choice` would logically make sense but we'd then have to convert it to a `Mask<uint64_t>` to perform the swap so I changed `iswap` arg to a `Mask<uint64_t>`